src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -687,4 +691,34 @@
       is_expected.to redirect_to(project_show_path(user.home_project))
     end
   end
+
+  describe 'POST #new_incident_request' do
+    before do
+      login user
+      request.env["HTTP_REFERER"] = root_url # Needed for the redirect_to :back
+    end
+
+    it "without an existent project will raise an exception" do
+      expect { post :new_incident_request, project: 'non:existent:project' }.to raise_error Project::UnknownObjectError
```

Comment:
For sure it will cause a 500, but do we want to avoid it generally speaking?

I mean, all the actions with the before_filter `set_project` or `set_project_by_id` raises an ActiveRecord::RecordNotFound exception and those are not rescued too, so 500:

```ruby 
before_filter :set_project, only: [:autocomplete_packages, :autocomplete_repositories, :users, :subprojects,
:new_package, :new_package_branch, :incident_request_dialog, :release_request_dialog,
:show, :linking_projects, :add_person, :add_group, :buildresult, :delete_dialog,
:destroy, :remove_path_from_target, :rebuild_time, :packages_simple,
:requests, :save, :monitor, :toggle_watch, :meta,
:prjconf, :edit, :save_comment, :edit_comment,
:status, :maintained_projects,
:add_maintained_project_dialog, :add_maintained_project, :remove_maintained_project,
:maintenance_incidents, :unlock_dialog, :unlock, :save_person, :save_group, :remove_role,
:move_path, :save_prjconf, :clear_failed_comment]

before_filter :set_project_by_id, only: [:update]
```

Should we cover this action with a before_filter handling the exception properly? or should I refactor only this action to handle the case of a non existent project? Have in mind that it only will occur if you "write" a non existent project (I'm not sure if this is deleted it will find it or not).

BTW, using `before_filter` just to load data is wrong in my opinion... and in the case of raising exceptions... even more.
http://blog.thefrontiergroup.com.au/2014/02/before_action_an_anti_pattern/